File: webapi/Classes/Seat.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using webapi.Classes;

namespace webapi.Classes
{
    public class Seat
    {

        [Key]
        [Required]
        public Guid Id { get; set; }
        public string Name {  get; set; }
        public string Zone {  get; set; }
        public string SeatNumber {  get; set; }
        public List<SeatType> SeatType {  get; set; }
        [ForeignKey("Location")] public Guid LocationId {  get; set; }
        public Location Location {  get; set; }
        public int Capacity {  get; set; }
    }

    public enum SeatType
    {
        Regular,
        VIP,
        ReducedVisibility,
        Accessible,
        StandingOnly
    }
}

File: angularapp/src/interfaces/seat.ts
import { Location } from "./location";

export interface Seat {
  id: string;
  name: string;
  zone: string;
  seatNumber: string;
  seatType: SeatType[];
  locationId: string;
  location: Location;
  capacity: number;
}
export enum SeatType {
  Regular,
  VIP,
  ReducedVisibility,
  Accessible,
  StandingOnly
}
